qiskit_experiments/test/iq_pulse_backend.py

```diff
@@ -0,0 +1,444 @@
+                "meas_freq_est": [0],
+                "buffer": 0,
+                "pulse_library": [],
+                "cmd_def": [],
+            }
+        )
+        self._rng = np.random.default_rng(0)
+        self.converter = None
+        self.logical_levels = None
+        self.noise = static_dissipators is not None
+
+        self.solver_method = solver_method
+
+        self.static_hamiltonian = static_hamiltonian
+        self.hamiltonian_operators = hamiltonian_operators
+        self.static_dissipators = static_dissipators
+        self.solver = Solver(
+            static_hamiltonian=self.static_hamiltonian,
+            hamiltonian_operators=self.hamiltonian_operators,
+            static_dissipators=self.static_dissipators,
+            **kwargs,
+        )
+        self._target = Target(dt=dt, granularity=16)
+
+        self.model_dim = self.solver.model.dim
+        if self.noise:
+            self.model_dim = self.model_dim**2
+        self.gound_state = np.zeros(self.model_dim)
+        self.gound_state[0] = 1
+        self.y_0 = np.eye(self.model_dim)
+        self._simulated_pulse_unitaries = {}
+
+    @property
+    def target(self):
+        return self._target
+
+    @property
+    def max_circuits(self):
+        return None
+
+    def defaults(self):
+        """return backend defaults"""
+        return self._defaults
+
+    @classmethod
+    def _default_options(cls):
+        return Options(shots=4000)
+
+    @property
+    def default_pulse_unitaries(self) -> Dict[Tuple, np.array]:
+        """Return the default unitary matrices of the backend."""
+        return self._simulated_pulse_unitaries
+
+    @default_pulse_unitaries.setter
+    def default_pulse_unitaries(self, unitaries: Dict[Tuple, np.array]):
+        """Set the default unitary pulses this allows the tests to simulate the pulses only once."""
+        self._simulated_pulse_unitaries = unitaries
+
+    @staticmethod
+    def _get_info(instruction: CircuitInstruction) -> Tuple[Tuple[int], Tuple[float], str]:
+        p_dict = instruction.operation
+        qubit = tuple(int(str(val)[-2]) for val in instruction.qubits)
+        params = tuple(float(val) for val in p_dict.params)
+        return qubit, params, p_dict.name
+
+    def iq_data(
+        self,
+        probability: np.ndarray,
+        shots: int,
+        centers: List[Tuple[float, float]],
+        width: float,
+        phase: Optional[float] = None,
+    ) -> Tuple[List, List]:
+        """Generates IQ data for each logical levels
+
+        Parameters
+        ----------
+        probability : np.ndarray
+            probability of occupation
+        shots : int
+            Number of shots
+        centers : List[Tuple[float, float]]
+            The central i and q points for each level
+        width : float
+            Width of IQ data distribution
+        phase : float, optional
+            Phase of IQ data, by default 0
+
+        Returns
```

Comment:
Please align the doc strings with the style of Qiskit experiments which looks like this:
```suggestion
        Args:
            probability : probability of occupation
            shots : Number of shots
            centers : The central i and q points for each level
            width : Width of IQ data distribution
            phase : Phase of IQ data, by default 0

        Returns:
```

Reply:
The type hnits in the docs are not needed when we have them in the function definition.